### PR TITLE
create a function project in intellj idea.

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/AzureFunctionsUtils.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            final File mvnCmd = mvnCmds.get(0);
+            final File folder = Files.createTempDir();
+            final Map<String, String> maps = new LinkedHashMap<>();
+            maps.put("-DarchetypeGroupId", "com.microsoft.azure");
+            maps.put("-DarchetypeArtifactId", "azure-functions-archetype");
+            maps.put("-DgroupId", groupId);
+            maps.put("-DartifactId", artifactId);
+            maps.put("-Dversion", version);
+            maps.put("-Dpackage", packageName);
+            maps.put("-DappName", appName);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/AzureFunctionsUtils.java]: `appName` has been removed from function archetype, the default value is `${artifactId}-${timestamp}`
`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/IntellijCommandUtils.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+public class IntellijCommandUtils {
+
+    public static String[] executeMultipleLineOutput(final String cmd, File cwd, final boolean isWindows)
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/IntellijCommandUtils.java]: remove `isWindows` from parameters? As get os could be an static method

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/facet/FunctionsFacetType.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.jetbrains.annotations.Nullable;
+
+public class FunctionsFacetType extends FacetType<FunctionsFacet, FunctionsFacetConfiguration> {
+    FunctionsFacetType() {
+        super(FunctionsFacet.FACET_TYPE_ID, "AzureFunctions", "AzureFunctions");
+    }
+
+    @Override
+    public FunctionsFacetConfiguration createDefaultConfiguration() {
+        return null;
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/facet/FunctionsFacetType.java]: to be implemented?

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/facet/FunctionsFacetType.java]: No, we don't have any customization for FunctionsFacetConfiguration. see FunctionsFacetConfigurationImpl.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/module/FunctionTriggerChooserStep.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Override
+    public void updateDataModel() {
+        wizardContext.putUserData(AzureFunctionsConstants.WIZARD_TRIGGERS_KEY, getSelectedTriggers().toArray(new String[0]));
+    }
+
+    @NotNull
+    private List<String> getSelectedTriggers() {
+        final DefaultTableModel model = (DefaultTableModel) table.getModel();
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/module/FunctionTriggerChooserStep.java]: make model as `Member Variable`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/module/FunctionTriggerChooserStep.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        final JPanel panel = new JPanel(new BorderLayout());
+        panel.add(builder.getPanel(), "North");
+        return panel;
+    }
+
+    private void setupFunctionTriggers() {
+        final DefaultTableModel model = (DefaultTableModel) table.getModel();
+        boolean isFirst = true;
+        for (final String s : Arrays.asList("HttpTrigger", "BlobTrigger", "QueueTrigger", "TimerTrigger", "EventHubTrigger")) {
+            model.addRow(new Object[] { isFirst, s });
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/module/FunctionTriggerChooserStep.java]: I think we could change this line to `model.addRow(new Object[] { StringUtils.equals("HttpTrigger",s), s });`, to show HttpTrigger is the default trigger/

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/module/FunctionTriggerChooserStep.java (hunk 4 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/wizards/functions/module/FunctionTriggerChooserStep.java]: No need

**Pull request merged**